### PR TITLE
Feature/to cache pipeline

### PR DESCRIPTION
**Pull request by Mee-gu**
【BugFix】release the textures before cleaning the vertor

### CODE REVIEW

File: src/backend/BindGroup.cpp [hunk 1]
```diff
@@ -55,6 +55,10 @@ BindGroup::TextureInfo& BindGroup::TextureInfo::operator=(TextureInfo&& rhs)
         rhs.retainTextures();
         releaseTextures();
         textures = rhs.textures;
+        
+        //release the textures before clearning the vertor
+        for (auto& texture : rhs.textures)
+            CC_SAFE_RELEASE(texture);
```
Comment (minggo) [inline src/backend/BindGroup.cpp]: The destruction will release the textures, why need to release it here?

Comment (Mee-gu) [reply src/backend/BindGroup.cpp]: It will be empty after you clean the texture vector, thus skip to release the textures

Comment (minggo) [reply src/backend/BindGroup.cpp]: Yep, can use `releaseTextures()` to remove duplicated code.

Comment (Mee-gu) [reply src/backend/BindGroup.cpp]: got it

**Comment by Mee-gu** [comment]
【Feature】use releaseTextures() to remove duplicated code and make the code looks clean

**Pull request merged**